src/document/doc-builders.js

```diff
@@ -56,7 +56,8 @@ function indent(contents) {
 }
 
 /**
- * @param {number} n
+ * TODO: add missing types for paramter `n`
```

Comment:
I am not convinced about this TODO. After searching through the source code, I could only see the following types used for `n`: `number`, `string`, `Doc` ... and I only see `Doc` type used in one place for `n`.

And I only see the same thing for the `makeAlign` function.

I had initially missed the `Doc` type option since it was ignored by a `@ts-ignore` comment. I just did another code search and do not find any other `@ts-ignore` comments related to the `align` function call.

Reply:
You fogot one fact, it's public.

Reply:
This is why we can't change `!n` to `n === 0` or `n === false`

Reply:
> it's public

as documented here, for the sake of extra clarity:

- https://github.com/prettier/prettier/blob/master/commands.md#align

I am now thinking this part of the documentation could use a little improvement, will take another look sometime.

Reply:
I didn't see the doc, so the doc says `number | string`, but we use `{type: "root"}` ourself...I don't know what to do.

Reply:
> we use `{type: "root"}` ourself

here (to be extra-clear): https://github.com/prettier/prettier/blob/2.0.5/src/document/doc-builders.js#L104-L106

and we know that `{type: "root"}` would satisfy the requirements to be a `Doc` object, which I had defined maybe too loosely here: https://github.com/prettier/prettier/blob/2.0.5/src/document/doc-builders.js#L4-L11

So we could even do this (does not give me type error), though I would probably not recommend it:

```diff
- * @param {number | string | Doc} n
+ * @param {number | Doc} n
```

I would favor that we keep this PR as-is, with the TODO comment, and see if we can improve it someday later.

Reply:
I think it's not `Doc`, just a special object with `root: true` let's use `number | string`, keep it same as the  documentation, and restore the one in `markAsRoot`(use  ts-ignore). Make sense?

Reply:
Agreed, I was thinking the same thing about the type.